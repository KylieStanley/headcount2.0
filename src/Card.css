.card {
  align-items: center;
  border: 1px solid black;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  margin: 5px;
  padding: 8px;
  width: 250px;
}

ul {
  list-style-type: none;
  padding: 0;
}

h2 {
  text-align: center;
}

.greater-than {
  font-weight: bold;
  color: blue;
}

.less-than {

}