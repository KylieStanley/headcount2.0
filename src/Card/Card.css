.selected { 
  box-shadow: 0 0 13px 3px #000000ad;
}


.card {
  align-items: center;
  border: 1px solid black;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  height: 340px;
  margin: 5px;
  padding: 20px 30px;
  width: 250px;
}

ul {
  list-style-type: none;
  padding: 0;
}

h2 {
  margin: 0;
  text-align: center;
}

.greater-than {
  font-weight: bold;
  color: blue;
}

.less-than {

}