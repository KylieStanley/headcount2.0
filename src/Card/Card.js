import React from 'react';
import './Card.css';

const Card = ({ selectCard, district, selectedCards}) => {
  let selectedStyle = 'blue'

  return (
    <div className={`card ${selectedStyle}`} onClick={selectCard.bind(null, district)}>
      <h2>{district.location}</h2>
      <ul>
      {
        Object.keys(district.stats).map((item, i) => {
          let style = district.stats[item] > .5 ? 'greater-than' : 'less-than';
          return <li className={style} key={i}>{item}: {district.stats[item]}</li>
        })
      }
      </ul>
    </div>
  );
}

export default Card;
