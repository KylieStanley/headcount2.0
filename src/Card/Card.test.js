import React from 'react';
import Data from '../data/kindergartners_in_full_day_program.js';
import Card from './Card.js';
import {shallow} from 'enzyme';


describe('Card', () => {
  it('should match the snapshot with data passed in correctly', () => {
    const wrapper = shallow( <Card
      district={{location: "COLORADO",
                  stats: {2004: .5}
                }}
    />);

    expect(wrapper).toMatchSnapshot();
  })

  it('should have a class of less-than', () => {
    const wrapper = shallow( <Card 
      district={{location: "COLORADO",
                  stats: {2004: 0.24}
                }}
    />);

    expect(wrapper.find('li').hasClass('less-than'));
  });

  it('should have a class of greater-than', () => {
    const wrapper = shallow( <Card 
      district={{location: "COLORADO",
                  stats: {2004: 0.54}
                }}
    />);
    
    expect(wrapper.find('li').hasClass('greater-than'));
  });
});