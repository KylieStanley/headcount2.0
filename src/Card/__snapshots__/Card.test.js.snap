// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    district={
        Object {
            "location": "COLORADO",
            "stats": Object {
              "2004": 0.5,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          COLORADO
</h2>,
        <ul>
          <li
                    className="less-than"
          >
                    2004
                    : 
                    0.5
          </li>
</ul>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="less-than"
>
              2004
              : 
              0.5
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2004",
                ": ",
                0.5,
              ],
              "className": "less-than",
            },
            "ref": null,
            "rendered": Array [
              "2004",
              ": ",
              0.5,
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            COLORADO
</h2>,
          <ul>
            <li
                        className="less-than"
            >
                        2004
                        : 
                        0.5
            </li>
</ul>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="less-than"
>
                2004
                : 
                0.5
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2004",
                  ": ",
                  0.5,
                ],
                "className": "less-than",
              },
              "ref": null,
              "rendered": Array [
                "2004",
                ": ",
                0.5,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
